packages/lib/gpcircuits/circuits/tuple.circom

```diff
@@ -0,0 +1,64 @@
+pragma circom 2.1.8;
+
+include "circomlib/circuits/poseidon.circom"; // Poseidon hash
+include "gpc-util.circom"; // input selector
+
+/**
+ * Module mapping MAX_VALUES signals to MAX_TUPLES tuples of
+ * arity TUPLE_ARITY, each of which is encoded in the form of a
+ * Poseidon hash on MAX_ARITY elements. Lower-order tuples
+ * are assumed to have 0s in unoccupied slots, which may
+ * be accomplished by setting the corresponding tuple index
+ * to a sufficiently large number, e.g. -1 mod p.
+ */
+
+template TupleModule(
+    // Arity of each generated tuple
+    TUPLE_ARITY,
+    // Maximum number of value hashes to choose from
+    MAX_VALUES,
+    // Maxium number of tuples to generate
+    MAX_TUPLES
```

Comment:
This is interesting.  What's the value of generating multiple tuples from a single module vs. having multiple tuple modules?

Reply:
Or maybe a MultiTupleModule template to deal with the routing of inputs to outputs between several TupleModule templates in a row.  I think some sort of hierarchical organization would be clearer than the multi-dimensional arrays here.